### PR TITLE
Set up firebase encoders in transport-runtime

### PR DESCRIPTION
**Pull request by James201311**
Set up firebase encoders in transport-runtime, it will be used to generate builder class and provide encoding methods for client metrics proto file.

### CODE REVIEW

File: transport/transport-runtime/code-gen-cfg.textproto [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# proto-file: firebase-android-sdk/transport/transport-runtime/src/main/proto/client_analytics.proto
+# proto-message: ClientMetrics
+
+vendor_package: "com.google.android.datatransport.runtime"
+include: "firebase.transport.ClientMetrics"
```
Comment (vkryachko) [inline transport/transport-runtime/code-gen-cfg.textproto:5]: nit: add new line at the end

File: transport/transport-runtime/transport-runtime.gradle [hunk 2]
```diff
@@ -47,6 +79,11 @@ android {
         targetCompatibility JavaVersion.VERSION_1_8
     }
     sourceSets {
+        main {
+            proto {
+                srcDir 'src/main/proto'
```
Comment (vkryachko) [inline transport/transport-runtime/transport-runtime.gradle:84]: why is this needed? afaict it should be the default even if you don't specify it

Comment (James201311) [reply transport/transport-runtime/transport-runtime.gradle:84]: Adding the above line will stop the android studio compiling about the syntax error.

Comment (vkryachko) [reply transport/transport-runtime/transport-runtime.gradle:84]: sgtm

**Review by vkryachko — APPROVED** [review]
